README.md

```diff
@@ -117,6 +118,69 @@ npm install --save react-a11y-dialog
 - **Default value**: `dialog`
 - **Description**: The `role` attribute of the dialog element, either `dialog` (default) or `alertdialog` to make it a modal (preventing closing on click outside of <kbd>ESC</kbd> key).
 
+## Hook
+
+The library exports both `A11yDialog`, a React component rendering a dialog while performing the `a11y-dialog` bindings under the hood, and a `useA11yDialog` hook providing only the binding logic without any markup.
+
+Using the hook can be handy when building your own dialog. Beware though, **it is an advanced feature**. Make sure to [stick to the expected markup](http://edenspiekermann.github.io/a11y-dialog/#expected-dom-structure).
+
+```js
+import { useA11yDialog } from 'react-a11y-dialog'
+
+const MyCustomDialog = props => {
+  // `instance` is a React ref containing the `a11y-dialog` instance.
+  // `attr` is an object with the following keys:
+  // - `container`: the dialog container
+  // - `overlay`: the dialog overlay (sometimes called backdrop)
+  // - `dialog`: the actual dialog box
+  // - `inner`: the inner dialog container
+  // - `title`: the dialog mandatory title
+  // - `closeButton`:  the dialog close button
+  const [instance, attr] = useA11yDialog({
```

Comment:
I'd maybe rename `attr` in the example to a more fitting name. Since this is mostly just for CSS classNames, why not `cssClasses`?

Reply:
Nevermind, I now realise it comes with more props as well.

Reply:
With all due respect, I believe you are mistaken about what this object contain. It contains all the props that need to make their way into the DOM, such as `id`, `role`, `onClick`, `tabIndex`…

It is this object: https://github.com/HugoGiraudel/react-a11y-dialog/pull/46/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R46-R54

Reply:
Yes, I realised as well :D That's why I marked my comment as resolved